### PR TITLE
WIP: Feature/lead confidentiality

### PR DESCRIPTION
**Pull request by bewakes**

### CODE REVIEW

File: apps/lead/migrations/0025_auto_20190912_0420.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('lead', '0024_lead_author'),
+    ]
+
+    operations = [
+        migrations.RunPython(
+            forward_merge_restricted_and_protected,
+            backward_do_nothing_just_for_compatibility,
```
Comment (thenav56) [inline apps/lead/migrations/0025_auto_20190912_0420.py]: You can also use
```suggestion
            reverse_code=migrations.RunPython.noop,
```

File: apps/project/admin.py [hunk 2]
```diff
@@ -84,4 +87,48 @@ class ProjectStatusAdmin(admin.ModelAdmin):
...
+            PROJECT_PERMISSIONS.export,
+        )
+        self.fields['assessment_permissions'].widget = PermissionsWidget(
+            'assessment_permissions',
+            PROJECT_PERMISSIONS.assessment,
+        )
+
+    def save(self, commit=True):
+        obj = super().save(commit=False)
+        print(self.cleaned_data)
```
Comment (thenav56) [inline apps/project/admin.py]:
```suggestion
```

File: apps/project/widgets.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            for k, v in self.permission_values.items()
+        ]
+        return reduce(
+            lambda acc, x: acc | x,
+            to_or_vals
+        )
+
+    def render(self, name, value, attrs=None, renderer=None):
+        html = ''
+        print(name, value)
```
Comment (thenav56) [inline apps/project/widgets.py]:
```suggestion
```

File: apps/project/widgets.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
+from functools import reduce
+
+from django import forms
+
+
+class PermissionsWidget(forms.Widget):
+    def __init__(self, name, permission_values, *args, **kwargs):
+        print(args, kwargs)
```
Comment (thenav56) [inline apps/project/widgets.py]:
```suggestion
```

File: apps/lead/migrations/0025_auto_20190912_0420.py (hunk 1 above)
Comment (bewakes) [reply apps/lead/migrations/0025_auto_20190912_0420.py]: done

File: apps/project/admin.py (hunk 2 above)
Comment (bewakes) [reply apps/project/admin.py]: done

File: apps/project/widgets.py (hunk 3 above)
Comment (bewakes) [reply apps/project/widgets.py]: done

File: apps/project/widgets.py (hunk 4 above)
Comment (bewakes) [reply apps/project/widgets.py]: done

**Review by thenav56 — APPROVED** [review]

**Pull request merged**